doesn't compile on nightly without std

Hi,
I tried adding `ethereum-types = { version = "0.3", default-features = false }` to my Cargo.toml, and it failed compiling:
```
   Compiling ethereum-types v0.3.2
error: cannot find macro `format!` in this scope
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/ethereum-types-0.3.2/src/hash.rs:72:1
   |
72 | construct_hash!(H32, 4);
   | ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: cannot find macro `format!` in this scope
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/ethereum-types-0.3.2/src/hash.rs:73:1
   |
73 | construct_hash!(H64, 8);
   | ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: cannot find macro `format!` in this scope
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/ethereum-types-0.3.2/src/hash.rs:74:1
   |
74 | construct_hash!(H128, 16);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: cannot find macro `format!` in this scope
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/ethereum-types-0.3.2/src/hash.rs:75:1
   |
75 | construct_hash!(H160, 20);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: cannot find macro `format!` in this scope
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/ethereum-types-0.3.2/src/hash.rs:76:1
   |
76 | construct_hash!(H256, 32);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: cannot find macro `format!` in this scope
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/ethereum-types-0.3.2/src/hash.rs:77:1
   |
77 | construct_hash!(H264, 33);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: cannot find macro `format!` in this scope
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/ethereum-types-0.3.2/src/hash.rs:78:1
   |
78 | construct_hash!(H512, 64);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: cannot find macro `format!` in this scope
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/ethereum-types-0.3.2/src/hash.rs:79:1
   |
79 | construct_hash!(H520, 65);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: cannot find macro `format!` in this scope
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/ethereum-types-0.3.2/src/hash.rs:80:1
   |
80 | construct_hash!(H1024, 128);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error[E0412]: cannot find type `String` in this scope
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/ethereum-types-0.3.2/src/hash.rs:72:1
   |
72 | construct_hash!(H32, 4);
   | ^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error[E0412]: cannot find type `String` in this scope
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/ethereum-types-0.3.2/src/hash.rs:73:1
   |
73 | construct_hash!(H64, 8);
   | ^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error[E0412]: cannot find type `String` in this scope
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/ethereum-types-0.3.2/src/hash.rs:74:1
   |
74 | construct_hash!(H128, 16);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error[E0412]: cannot find type `String` in this scope
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/ethereum-types-0.3.2/src/hash.rs:75:1
   |
75 | construct_hash!(H160, 20);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error[E0412]: cannot find type `String` in this scope
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/ethereum-types-0.3.2/src/hash.rs:76:1
   |
76 | construct_hash!(H256, 32);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error[E0412]: cannot find type `String` in this scope
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/ethereum-types-0.3.2/src/hash.rs:77:1
   |
77 | construct_hash!(H264, 33);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error[E0412]: cannot find type `String` in this scope
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/ethereum-types-0.3.2/src/hash.rs:78:1
   |
78 | construct_hash!(H512, 64);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error[E0412]: cannot find type `String` in this scope
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/ethereum-types-0.3.2/src/hash.rs:79:1
   |
79 | construct_hash!(H520, 65);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error[E0412]: cannot find type `String` in this scope
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/ethereum-types-0.3.2/src/hash.rs:80:1
   |
80 | construct_hash!(H1024, 128);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: aborting due to 18 previous errors

For more information about this error, try `rustc --explain E0412`.
error: Could not compile `ethereum-types`.

To learn more, run the command again with --verbose.
```

My rust version:
```
root@197754312d3b:~# rustc --version
rustc 1.27.0-nightly (ad610bed8 2018-04-11)
root@197754312d3b:~# cargo --version
cargo 1.26.0-nightly (b70ab13b3 2018-04-04)
```

--- a/ethbloom/Cargo.toml
+++ b/ethbloom/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "ethbloom"
-version = "0.5.0"
+version = "0.5.1"
 authors = ["Parity Technologies <admin@parity.io>"]
 description = "Ethereum bloom filter"
 license = "MIT"

--- a/ethereum-types/Cargo.toml
+++ b/ethereum-types/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "ethereum-types"
-version = "0.3.2"
+version = "0.3.3"
 authors = ["Parity Technologies <admin@parity.io>"]
 license = "MIT"
 homepage = "https://github.com/paritytech/primitives"

